src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/BookingFacade.kt

```diff
@@ -0,0 +1,30 @@
+package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service
+
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.config.EmailService
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.entity.VideoBooking
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.request.CreateVideoBookingRequest
+
+/**
+ * This facade exists to ensure booking related transactions are fully committed prior to sending any emails.
+ */
+@Component
+class BookingFacade(
+  private val createVideoBookingService: CreateVideoBookingService,
+  private val emailService: EmailService,
+) {
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun create(booking: CreateVideoBookingRequest): Long =
```

Comment:
We may want to return some better model representation of the newly created booking but for now just returning the identifier.